espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ActivityHelper.java

```diff
@@ -32,4 +56,42 @@ public static Activity getCurrentActivity() throws AppiumException {
 
         throw new AppiumException("Failed to get current Activity");
     }
+
+    private static String getFullyQualifiedActivityName(Instrumentation mInstrumentation, String appActivity) {
+        return getFullyQualifiedActivityName(mInstrumentation.getTargetContext().getPackageName(), appActivity);
+    }
+
+    private static String getFullyQualifiedActivityName(String pkg, String appActivity) {
+        return appActivity.startsWith(".") ? pkg + appActivity : appActivity;
+    }
+
+    public static void startActivityViaInstruments(String appActivity, boolean waitForActivity) {
+        logger.info(String.format("Starting activity '%s'", appActivity));
+        Instrumentation mInstrumentation = InstrumentationRegistry.getInstrumentation();
+        Intent intent = new Intent(Intent.ACTION_MAIN);
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        String fullyQualifiedAppActivity = getFullyQualifiedActivityName(mInstrumentation, appActivity);
+        Instrumentation.ActivityMonitor mSessionMonitor = mInstrumentation
+                .addMonitor(fullyQualifiedAppActivity, null, false);
+        intent.setClassName(mInstrumentation.getTargetContext(), fullyQualifiedAppActivity);
+        mInstrumentation.startActivitySync(intent);
+        if (waitForActivity) {
```

Comment:
This is not in this PR thing, but I noticed that we should consider `appWaitActivity`.
For example,

1. launch homeActivity
2. Then, app shows some promotion activity
3. Appium should interact it (a user can specify the promotion activity as `appWaitActivity` in UIA2)

Currently, if this happens, this monitor never retun

Reply:
fixed

Reply:
Thanks! Can add the same logic in startActivityViaTestRule since `start_activity` also have `appWaitActivity` param

Reply:
I don't think there we can add it, since the lifecycle monitor is private to ActivityTestRule, so it can only wait for the activity, which class is set in the constructor